docs/cookbook/entities/custom-model-accessible-per-admin-channel.rst

```diff
@@ -0,0 +1,392 @@
+
+                    if ($adminChannel != null) {
+                        return [$adminChannel];
+                    }
+
+                    return $this->channelRepository->findAll();
+                },
+                'choice_value' => 'code',
+                'choice_label' => 'name',
+                'choice_translation_domain' => false,
+            ]);
+        }
+
+        public function getParent(): string
+        {
+            return ChoiceType::class;
+        }
+
+        public function getBlockPrefix(): string
+        {
+            return 'sylius_channel_choice';
+        }
+    }
+
+.. code-block:: yaml
+
+    App\Form\ChannelPerAdminChannelType:
+        arguments: ['@sylius.repository.channel', '@Sylius\Plus\ChannelAdmin\Application\Provider\AdminChannelProviderInterface']
+        tags: ['form.type']
+
+The ``Sylius\Plus\ChannelAdmin\Application\Provider\AdminChannelProviderInterface`` service allows getting a list of proper channels for the currently logged-in admin.
+
+* Next, create a form type for your entity:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Form;
+
+    use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
+    use Symfony\Component\Form\Extension\Core\Type\TextType;
+    use Symfony\Component\Form\FormBuilderInterface;
+
+    final class SupplierType extends AbstractResourceType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('name', TextType::class, [
+                    'label' => 'Name'
+                ])
+                ->add('channel', ChannelPerAdminChannelType::class, [
+                    'label' => 'Channel'
+                ])
+            ;
+        }
+
+        public function getBlockPrefix(): string
+        {
+            return 'supplier';
+        }
+    }
+
+.. code-block:: yaml
+
+    App\Form\ChannelPerAdminChannelType:
+            arguments: ['@sylius.repository.channel', '@Sylius\Plus\ChannelAdmin\Application\Provider\AdminChannelProviderInterface']
+            tags: ['form.type']
+
+Remember to register ``App\Form\ChannelPerAdminChannelType`` for resource:
+
+.. code-block:: yaml
+
+    sylius_resource:
+    resources:
+        app.supplier:
+            driver: doctrine/orm
+            classes:
+                model: App\Entity\Supplier
+                repository: App\Repository\SupplierRepository
+   +            form: App\Form\SupplierType
+
+
+1. Add your entity to admin menu:
+----------------------------------
+
+Adding a new entity to the admin menu is described in the section ``How to customize Admin Menu`` of :doc:`this guide </customization/menu>`.
+
+1. Restrict access to the entity for the respective channel administrator roles (using ACL/RBAC):
+-------------------------------------------------------------------------------------------------
```

Comment:
Is this point necessary in cookbook for admin channel?

Reply:
IMO is better to have. But

Reply:
IIRC channel admins and RBAC are different features, so we can mention RBAC but the configuration of the new RBAC role should be in a separate doc.